### PR TITLE
Improve random number handling.

### PR DESCRIPTION
**Pull request by stoeckmann**
Allocate PRNG only once (at least that's how I interpret the intention of the variable seed).
Avoid signed overflows in rare cases.

### CODE REVIEW

File: src/popt.c [hunk 1]
```diff
@@ -1024,9 +1030,11 @@ int poptSaveShort(short * arg, unsigned int argInfo, long aLong)
 	if (!seed) {
 	    srandom((unsigned)getpid());
 	    srandom((unsigned)random());
+	    seed = 1;
 	}
+	aLong = aLong < -LONG_MAX ? LONG_MAX : aLong;
 	aLong = random() % (aLong > 0 ? aLong : -aLong);
-	aLong++;
+	aLong = aLong == LONG_MAX ? 0 : aLong + 1;
```
Comment (pmatilai) [inline src/popt.c:1037]: Please use parenthesis for readability in cases like this, ie
```
aLong = (aLong == LONG_MAX) ? 0 : aLong + 1;
```

**Comment by pmatilai** [comment]
Hohum, this whole random thing looks like one big :hankey: - thanks for pointing it out.

The random stuff was silently added in commit 149f325ff8125f8454629da2a64f7677e9165c89 which claims to be about bringing rpm.org changes in, which is absolutely bogus - there never was such an addition in the old rpm.org popt codebase. And it's straight out of the copy-paste coding school text-book :roll_eyes: 

Let's do this right. Care to do a little refactoring to push the common logic into a function or two? (if not that's okay, I can do that). 

(It doesn't end there either, "avoiding i18n churn" seems like a silly reason to avoid adding a needed return code. OTOH the *need* for that seems dubious to me, srandom() is in POSIX.1-2001 and I'm not entirely sure we care beyond that, but these are beyond the scope I guess)

**Comment by stoeckmann** [comment]
I have moved the random handling into its own static function poptRandomValue. Less code to review, definitely. ;)

**Review by pmatilai — APPROVED** [review]
Yeah, much much nicer this way. Thanks!

**Pull request merged**